modules/engine/src/model/model.ts

```diff
@@ -295,7 +295,9 @@ export class Model {
     // TODO - delete previous geometry?
     this.vertexCount = gpuGeometry.vertexCount;
     this.setAttributes(gpuGeometry.attributes);
-    this.setIndexBuffer(gpuGeometry.indices);
+    if (gpuGeometry.indices) {
+      this.setIndexBuffer(gpuGeometry.indices);
```

Comment:
Just a minor concern: 
- I tried to design the API so that setIndexBuffer could take `null` so that indices can be unset. 
- Otherwise one may not be able to change from an indexed to a non-indexed geometry. 
- Also WebGPU one has to reset the buffers between every draw call but I think that is handled on a lower level.